File: CPP/Cpu.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"


using namespace std;

// Kelas Cpu yang mewarisi dari kelas Komponen
class Cpu : public Komponen
{
private:
    int jumlahCore; // atribut menyimpan jumlah core cpu
    float kecepatanGHz; // atribut menyimpan kecepatan cpu dalam GHz
public:
    Cpu()
    {


    }


    // Konstruktor dengan parameter untuk menginisialisasi jumlahCore, kecepatanGHz, merk, dan nama
    Cpu(int jumlahCore, float kecepatanGHz, string merk, string nama) : Komponen(merk, nama)
    {
        this->jumlahCore = jumlahCore;
        this->kecepatanGHz = kecepatanGHz;
    }

    //Setter
    void setJumlahCore(int jumlahCore)
    {
        this->jumlahCore = jumlahCore;
    }


    void setKecepatanGHz(float kecepatanGHz)
    {
        this->kecepatanGHz = kecepatanGHz;
    }


    //Getter
    int getJumlahCore()
    {
        return this->jumlahCore;
    }


    float getKecepatanGHz()
    {
        return this->kecepatanGHz;
    }


    ~Cpu()
    {


    }
};

File: python/Cpu.py
from Komponen import Komponen

class Cpu(Komponen):
    #atribut private
    __jumlah_core = ""
    __kecepatan_ghz = ""

    #constructor
    def __init__(self, jumlah_core, kecepatan_ghz, merk="", nama=""):
        super().__init__(merk, nama)
        self.__jumlah_core = jumlah_core
        self.__kecepatan_ghz = kecepatan_ghz

    #setter
    def set_jumlah_core(self, jumlah_core):
        self.__jumlah_core = jumlah_core

    def set_kecepatan_ghz(self, kecepatan_ghz):
        self.__kecepatan_ghz = kecepatan_ghz

    #getter
    def get_jumlah_core(self):
        return self.__jumlah_core

    def get_kecepatan_ghz(self):
        return self.__kecepatan_ghz